From cec7e136ce7d889cb8aca4b001230dad594ea1ed Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Fri, 23 Aug 2013 10:03:12 +0800
Subject: [PATCH 796/838] qman:add qman_release_fqid_all()

Add qman_release_fqid_all() routine to release all fq used by qman.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/staging/fsl_qbman/dpa_alloc.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/staging/fsl_qbman/dpa_alloc.c b/drivers/staging/fsl_qbman/dpa_alloc.c
index 44db3e1..b62b502 100644
--- a/drivers/staging/fsl_qbman/dpa_alloc.c
+++ b/drivers/staging/fsl_qbman/dpa_alloc.c
@@ -400,6 +400,16 @@ struct alloc_node {
 	int is_alloced;
 };
 
+void qman_release_fqid_all(void)
+{
+	struct alloc_node *p;
+
+	list_for_each_entry(p, &(fqalloc.used), list) {
+		fq_cleanup(p->base);
+	}
+}
+EXPORT_SYMBOL(qman_release_fqid_all);
+
 /* #define DPA_ALLOC_DEBUG */
 
 #ifdef DPA_ALLOC_DEBUG
-- 
2.0.2

